src/tiled/propertieswidget.cpp

```diff
@@ -143,9 +146,378 @@ static bool anyObjectHasProperty(const QList<Object*> &objects, const QString &n
+                     QObject *parent = nullptr)
+        : AbstractProperty(tr("Tile Size"), editorFactory, parent)
+        , mMapDocument(mapDocument)
+    {
+    }
+
+    QVariant value() const override
+    {
+        return mMapDocument->map()->tileSize();
+    }
+
+    void setValue(const QVariant &value) override
+    {
+        auto oldSize = mMapDocument->map()->tileSize();
+        auto newSize = value.toSize();
+
+        if (oldSize.width() != newSize.width()) {
+            auto command = new ChangeMapProperty(mMapDocument,
+                                                 Map::TileWidthProperty,
+                                                 newSize.width());
+            mMapDocument->undoStack()->push(command);
+        }
+
+        if (oldSize.height() != newSize.height()) {
+            auto command = new ChangeMapProperty(mMapDocument,
+                                                 Map::TileHeightProperty,
+                                                 newSize.height());
+            mMapDocument->undoStack()->push(command);
+        }
+    };
+
+private:
+    MapDocument *mMapDocument;
+};
+
+class MapProperties : public QObject
+{
+    Q_OBJECT
+
+public:
+    MapProperties(MapDocument *mapDocument,
+                  ValueTypeEditorFactory *editorFactory,
+                  QObject *parent = nullptr)
+        : QObject(parent)
+        , mMapDocument(mapDocument)
+        , mSizeProperty(new MapSizeProperty(mapDocument, editorFactory, this))
+        , mTileSizeProperty(new TileSizeProperty(mapDocument, editorFactory, this))
+    {
+        mClassProperty = editorFactory->createProperty(
+                    tr("Class"),
+                    [this]() {
+                        return map()->className();
+                    },
+                    [this](const QVariant &value) {
+                        push(new ChangeClassName(mMapDocument, { map() },
+                                                 value.toString()));
+                    });
+
+        mOrientationProperty = editorFactory->createProperty(
+                    tr("Orientation"),
+                    [this]() {
+                        return QVariant::fromValue(map()->orientation());
+                    },
+                    [this](const QVariant &value) {
+                        auto orientation = static_cast<Map::Orientation>(value.toInt());
+                        push(new ChangeMapProperty(mMapDocument, orientation));
+                    });
+
+        mInfiniteProperty = editorFactory->createProperty(
+                    tr("Infinite"),
+                    [this]() {
+                        return map()->infinite();
+                    },
+                    [this](const QVariant &value) {
+                        push(new ChangeMapProperty(mMapDocument,
+                                                   Map::InfiniteProperty,
+                                                   value.toInt()));
+                    });
```

Comment:
The new `GetSetProperty`, along with the `createProperty` helper function, is turning out to be quite near to the minimum code currently needed to expose these properties in the UI. :-)

Reply:
That seems lighter indeed. Nice